File: PRWAS.cpp
#include <iostream>
using namespace std;
#define MOD 1000000007
int main(){
      ios_base::sync_with_stdio(0);
     int T,N;
     long long sum,temp1,temp2,temp3,flag=0,temp4,pt=14;
     cin >> T;
     while(T--){
     	cin >> N;
     	if(N==1) cout << "1\n";
        else{
        	temp1=N+1;
            temp2=N+temp1;
            if(!(N%3)){N/=3;flag++;}
            else if(!(temp1%3)){temp1/=3;flag++;}
            else temp2/=3;

            temp1=(temp1*N)%MOD;
            if(flag) pt*=3;
            temp3=pt*temp1;
            temp4=(16*N)%MOD;
        	while(true){
            
        }
      }
  }
}

File: PRWAS.py
num=input()
while num>0:
	n=input()
	if n==1:
		print("%d\n"%n)
	else:
		temp1=(n*(n+1))/2;
		temp2=(n<<1)+1;
		#print temp1, temp2
		total=4*((4*(temp1*temp2)/3)-(7*temp1)+4*n)
		print (total-3)%1000000007
	num-=1;